src/request/mod.rs

```diff
@@ -1,105 +1,42 @@
-//! Module containing all structs modelling requests to the boomlings APIs.
+//! Module containing all structs modelling requests to the Boomlings APIs.
 //!
 //! These directly implement (de)serialization into RobTop's data format, unlike models where
 //! RobTop's eccentricities are hidden. This is since directly re-using these structs outside of
 //! making/proxying requests for the boomlings servers seems rather useless to me, as they already
 //! contain a lot of boomlings-specific fields.
+//! This can also be edited for a specific GDPS, e.g 1.9 GDPS. (hi absowute :3)
 
 use crate::{model::GameVersion, serde::RequestSerializer};
 use serde::{Deserialize, Serialize};
+use once_cell::sync::OnceCell;
 
-macro_rules! const_setter {
-    ($name: ident, $field: ident, $t: ty) => {
-        pub const fn $name(mut self, $field: $t) -> Self {
-            self.$field = $field;
-            self
-        }
-    };
-
-    ($name: ident, $t: ty) => {
-        pub const fn $name(mut self, arg0: $t) -> Self {
-            self.$name = arg0;
-            self
-        }
-    };
+static GDPS_URL: OnceCell<String> = OnceCell::new();
 
-    ($(#[$attr:meta])* $name: ident: $t: ty) => {
-        $(#[$attr])*
-        pub const fn $name(mut self, $name: $t) -> Self {
-            self.$name = $name;
-            self
-        }
-    };
-
-    ($(#[$attr:meta])* $field:ident[$name: ident]: $t: ty) => {
-        $(#[$attr])*
-        pub const fn $name(mut self, $field: $t) -> Self {
-            self.$field = $field;
-            self
-        }
-    }
+pub fn get_gdps_url() -> String {
+    GDPS_URL.get_or_init(|| std::env::var("GDPS_URL").unwrap_or_else(|_| "https://www.boomlings.com/database/".to_string()))
```

Comment:
No need for the environment variable anymore now
```suggestion
    GDPS_URL.get_or_init(|| BOOMLINGS_REQUEST_BASE)
```